File: java/ecbenchmark/src/ecbenchmark/cec10/F20.java
/*
 * To change this template, choose Tools | Templates
 * and open the template in the editor.
 */
package ecbenchmark.cec10;

import ecbenchmark.function.Rosenbrock;
import ecbenchmark.Function;
import ecbenchmark.operator.Shifted;

/**
 *
 * @author jcrada
 */
public class F20 extends CecFunction {

    private Function rosenbrock;

    public F20(int dimensions, int mValue) {
        super("F20", "Shifted Rosenbrock's Function", dimensions,
                -100, 100, true, new CecRandom(20l), mValue);

        double[] shift = CecMath.ShiftVector(dimensions, getMinimum(), getMaximum() - 1d, getRandomizer());
        rosenbrock = new Shifted(shift, 0, new Rosenbrock());
    }

    public double f(final double[] x) {
        return rosenbrock.f(x);
    }
}

File: cpp/ecbenchmark/src/cec10/F20.cpp
#include "ecbenchmark/cec10/F20.h"
#include "ecbenchmark/cec10/CecMath.h"
#include "ecbenchmark/operator/Shifted.h"
#include "ecbenchmark/function/Rosenbrock.h"

namespace ecb {
    namespace cec10 {

        F20::F20(int dimensions, int mValue)
        : CecFunction("F20", "Shifted Rosenbrock's Function", dimensions,
        -100, 100, true, mValue, new CecRandom(20l)) {

            std::vector<scalar>  shift = CecMath::ShiftVector(dimensions, minimumDomain(), maximumDomain() - 1, randomizer());
            rosenbrock = new Shifted(shift, 0, new Rosenbrock());

        }

        F20::~F20() {
            delete rosenbrock;
        }

        scalar F20::f(const std::vector<scalar>& x) {
            return rosenbrock->f(x);
        }
    }
}

